front/src/modules/simulationResult/components/SpaceTimeChartWrapper/useTrackOccupancy.ts

```diff
@@ -162,35 +196,136 @@ const useTrackOccupancy = ({
+      const pacedIds: PacedTrainId[] = [];
+      for (const id of Object.keys(trainsCollection)) {
+        if (isTrainScheduleId(id)) scheduleIds.push(id);
+        else if (isPacedTrainId(id)) pacedIds.push(id);
+      }
+
+      const bodyForTrainSchedules =
+        scheduleIds.length > 0
+          ? {
+              operational_point_id: opId,
+              infra_id: infraId,
+              train_schedule_ids: scheduleIds.map(extractEditoastIdFromTrainScheduleId),
+            }
+          : null;
+
+      const bodyForPaced =
+        pacedIds.length > 0
+          ? {
+              operational_point_id: opId,
+              infra_id: infraId,
+              paced_train_ids: pacedIds.map(extractEditoastIdFromPacedTrainId),
+            }
+          : null;
+
+      const [trainScheduleResp, pacedResp] = await Promise.all([
+        bodyForTrainSchedules
+          ? postTrainScheduleTrackOccupancy({ body: bodyForTrainSchedules })
+          : Promise.resolve(undefined),
+        bodyForPaced
+          ? postPacedTrainTrackOccupancy({ body: bodyForPaced })
+          : Promise.resolve(undefined),
+      ]);
+
+      const zones: MovableOccupancyZone[] = [];
+
+      if (trainScheduleResp?.data) {
+        for (const [trackId, occupations] of Object.entries(trainScheduleResp.data)) {
+          for (const occupation of occupations) {
+            const trainId = formatEditoastIdToTrainScheduleId(occupation.train_schedule_id);
+            const train = trainsCollection[trainId];
+            if (!train) continue;
+
+            zones.push(
+              getMovableOccupancyZone(
+                trackId,
+                trainId,
+                occupation,
+                train.spaceTimeCurves,
+                train.name,
+                train.departureTime
+              )
+            );
+          }
+        }
+      }
+
+      if (pacedResp?.data) {
+        for (const [trackId, occupations] of Object.entries(pacedResp.data)) {
+          for (const occupation of occupations) {
+            const pacedId = formatEditoastIdToPacedTrainId(occupation.paced_train_id);
+            const train = trainsCollection[pacedId];
+            if (!train) continue;
+
+            if (!('paced' in train)) {
+              continue;
+            }
+
+            const occurrenceId =
+              occupation.type === 'CreatedException'
+                ? formatPacedTrainIdToExceptionId(pacedId, occupation.exception_key)
+                : formatPacedTrainIdToIndexedOccurrenceId(pacedId, occupation.index ?? 0);
+
+            const exception =
+              occupation.type === 'CreatedException'
+                ? train.exceptions.find((e) => e.key === occupation.exception_key)
+                : train.exceptions.find((e) => e.occurrence_index === (occupation.index ?? 0));
+
+            const spaceTimeCurves =
+              train.exceptionProjections.get(occurrenceId)?.spaceTimeCurves ??
+              train.spaceTimeCurves;
```

Comment:
You should get the exception projection with the key of the exception. Actually, the exception key is a bit like the exception id :)
You can do smth like this:
```suggestion
           const exceptionProjection =
              occupation.type !== 'BaseOccurrence'
                ? train.exceptionProjections.get(occupation.exception_key)
                : undefined;
            const { spaceTimeCurves } = exceptionProjection ?? train;
```